### PR TITLE
Implement TramplesMinimize as a move flag

### PR DESCRIPTION
**Pull request by Keyacom**
Implemented `TramplesMinimize` as a move flag. Due to it, removed effect codes `FlinchTargetTrampleMinimize` and `ParalyzeTargetTrampleMinimize` (the second one was exclusive to Body Slam). Moves that had them now have the effect codes `FlinchTarget` and `ParalyzeTarget`, respectively.

The code now does not check if the parameter of `tramplesMinimize?` is 1 or 2, it now checks if it is `true` or `false`. For the effect that causes the moves with this flag to skip accuracy checks, it also checks if `Settings::MECHANICS_GENERATION` is equal to or greater than 6.

Data on which moves are to be treated as able to double damage if Minimized and skip accuracy checks are [here](https://bulbapedia.bulbagarden.net/wiki/Minimize_(move)#Vulnerability_to_moves).

**Remarks:**
- Body Slam, Dragon Rush, Heat Crash and Heavy Slam could not trample Minimize in Gen 5. Body Slam, Dragon Rush and Heat Crash could do so in Gen 6+, but Heavy Slam could not until Gen 7.
- Double Iron Bash could only trample Minimize in Gen 7.

**TL;DR:**
- Implemented `TramplesMinimize` as a move flag.
- Modified the check for this move flag.
- Removed effect codes `FlinchTargetTrampleMinimize` and `ParalyzeTargetTrampleMinimize`. Moves that had them now have the effect codes `FlinchTarget` and `ParalyzeTarget`, respectively.

### CODE REVIEW

**Comment by Keyacom** [comment]
Please deny the commits "Remove an extra N from comment" (a663bc3) and "Merge branch 'dev' of https://github.com/Keyacom/pokemon-essentials i…" (03805f3)

This is because I forgot to copy the `dev` branch for this PR.

**Comment by Keyacom** [comment]
After you review this PR and accept changes, I will delete my fork and then re-fork this repo. I'll be more careful with where to make changes from then on.

**Comment by Maruno17** [comment]
Superseded by pull request #175.
